### PR TITLE
Cleanup monostub launcher code

### PR DESCRIPTION
**Pull request by Therzok**
* Removes a lot of `char *` <-> `NSString` conversions
* Reduces code duplication
* Removes unused defines
* Removes code which could simply invoke mono API

### CODE REVIEW

**Comment by Therzok** [comment]
cc @slluis

File: main/build/MacOSX/Makefile.am [hunk 1]
```diff
@@ -37,10 +35,10 @@ dmg: render.exe app
 	./make-dmg-bundle.sh
 
 monostub.o: monostub.mm $(MONOSTUB_EXTRA_SOURCEFILES)
-	g++ -g $(EXTERN_C_XM_REGISTRAR_DEFINE) $(MONOSTUB_DEFINES) $(HYBRID_SUSPEND_ABORT) -c -Wall -mmacosx-version-min=10.10 -m$(MONOSTUB_ARCH) -o $@ monostub.mm
+	g++ -g $(HYBRID_SUSPEND_ABORT) -c -Wall -mmacosx-version-min=10.10 -m$(MONOSTUB_ARCH) -o $@ monostub.mm
 
 monostub-nogui.o: monostub.mm $(MONOSTUB_EXTRA_SOURCEFILES)
-	g++ -g $(EXTERN_C_XM_REGISTRAR_DEFINE) $(MONOSTUB_DEFINES) $(HYBRID_SUSPEND_ABORT) -DNOGUI -c -Wall -mmacosx-version-min=10.10 -m$(MONOSTUB_ARCH) -o $@ monostub.mm
+	g++ -g $(HYBRID_SUSPEND_ABORT) -DNOGUI -c -Wall -mmacosx-version-min=10.10 -m$(MONOSTUB_ARCH) -o $@ monostub.mm
```
Comment (rodrmoya) [inline main/build/MacOSX/Makefile.am]: Is 10.10 really the minimum supported version? Didn’t we bump it recently to something newer?

Comment (Therzok) [reply main/build/MacOSX/Makefile.am]: We don't really use any symbols from 10.11 in the launcher code.

Comment (Therzok) [reply main/build/MacOSX/Makefile.am]: I'll bump to 10.11.

Comment (Therzok) [reply main/build/MacOSX/Makefile.am]: Fixed.

File: main/build/MacOSX/monostub-utils.h [hunk 2]
```diff
@@ -40,106 +38,35 @@ check_mono_version (const char *version, const char *req_version)
...
+		destinationOfSymbolicLinkAtPath:@"/var/db/xcode_select_link"
+		error:nil
+	];
 
-	free (lib_dir);
-	free (monodevelop_bin_dir);
-	free (value);
-	return result;
+	return xcode_link ? xcode_link : @"/Applications/Xcode.app/Contents/Developer";
 }
```
Comment (rodrmoya) [inline main/build/MacOSX/monostub-utils.h]:
```suggestion
return xcode_link ?: @"/Applications/Xcode.app/Contents/Developer"
```

**Comment by Therzok** [comment]
Just found out you can do:
```objc
char *x = "asdf";
NSString *a = @(x);
```

Will leave that for another time.

**Pull request merged**